tests/operators/test_agnostic_append.py

```diff
@@ -0,0 +1,116 @@
+import pathlib
+
+import pandas as pd
+from airflow.exceptions import BackfillUnfinished
+
+from astro import sql as aql
+from astro.dataframe import dataframe as adf
+from tests.operators import utils as test_utils
+
+CWD = pathlib.Path(__file__).parent
+import pytest
+
+
+@adf
+def validate_basic(df: pd.DataFrame):
+    assert len(df) == 6
+    assert not df["sell"].hasnans
+    assert df["rooms"].hasnans
+
+
+@adf
+def validate_append_all(df: pd.DataFrame):
+    assert len(df) == 6
+    assert not df["sell"].hasnans
+    assert not df["rooms"].hasnans
+
+
+@adf
+def validate_caste_only(df: pd.DataFrame):
+    assert len(df) == 6
+    assert not df["age"].hasnans
+    assert df["sell"].hasnans
+
+
+@pytest.fixture
+def append_params(request):
```

Comment:
I think this should be expanded to testcases instead of fixtures?

Reply:
I think this will make it much easier to add more testcases with less boilerplate. We can see if it becomes a problem but this ultimately makes it easier to create test grids

Reply:
I agree with @utkarsharma2 on this one. I don't think we should aim to have a single test per operator. We already have a few dimensions we are using fixtures:
- multiple databases, when relevant
- multiple files, when relevant
- multiple file locations, when relevant
- multiple file types, when relevant

I strongly recommend we do not use parametrization for groups of parameters sent to our tasks/operators.